rules/consistent-destructuring.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+const getDocsUrl = require('./utils/get-docs-url');
+
+const fix = (fixer, source, objectPattern, memberExpression) => {
+	const member = source.getText(memberExpression.property);
+	const {properties} = objectPattern;
+	const declarations = properties.map(property => source.getText(property));
+	const fixings = [fixer.replaceText(memberExpression, member)];
+
+	// Member is not already destructured
+	if (!declarations.includes(member)) {
+		const lastProperty = properties[properties.length - 1];
+
+		if (lastProperty) {
+			fixings.push(fixer.insertTextAfter(lastProperty, `, ${member}`));
+		} else {
+			fixings.push(fixer.replaceText(objectPattern, `{${member}}`));
+		}
+	}
+
+	return fixings;
+};
+
+const create = context => {
+	const source = context.getSourceCode();
+	const declarations = new Map();
+
+	return {
+		VariableDeclarator(node) {
+			const {id, init} = node;
+
+			if (id.type === 'ObjectPattern' && init.type !== 'Literal') {
```

Comment:
You can put both of these in the [selector](https://eslint.org/docs/developer-guide/selectors). Instead of `VariableDeclarator` it will be something like `'VariableDeclarator[id.type="ObjectPattern"][init.type!="Literal"]`.

Reply:
Yeah, I just thought it improved readability to have these in a separate `if` statement.